File: app/src/main/java/com/textfield/json/ottawabasketballcourts/Court.java
package com.textfield.json.ottawabasketballcourts;

import com.google.android.gms.maps.model.LatLng;

/**
 * Created by Jason on 20/02/2016.
 */
public class Court{
    private String name, type;
    private LatLng location;
    private int id;

    public Court(String type, String text, double lat, double lng, int id) {
        name = text;
        location = new LatLng(lat, lng);
        this.type = type;
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String getType() {
        return type;
    }

    public LatLng getLocation() {
        return location;
    }

    public String getName() {
        return name;
    }

    @Override
    public String toString() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: app/src/main/java/com/textfield/json/ottawabasketballcourts/Court.kt
package com.textfield.json.ottawabasketballcourts

import android.content.Context
import android.database.Cursor
import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.BitmapDescriptorFactory
import com.google.android.gms.maps.model.LatLng
import com.google.android.gms.maps.model.MarkerOptions
import java.util.*

class Court(context: Context, cursor: Cursor, googleMap: GoogleMap) {

    private val name = cursor.getString(cursor.getColumnIndex("name"))!!
    private val nameFr = cursor.getString(cursor.getColumnIndex("nameFr"))!!
    val type = cursor.getInt(cursor.getColumnIndex("type"))
    private val typeString: String = if (type == 1) {
        context.resources.getString(R.string.full)
    } else {
        context.resources.getString(R.string.half)
    }
    val id = cursor.getInt(cursor.getColumnIndex("id"))
    private val location: LatLng = LatLng(cursor.getDouble(cursor.getColumnIndex("latitude")), cursor.getDouble(cursor.getColumnIndex("longitude")))

    val marker = googleMap.addMarker(MarkerOptions().position(location)
            .icon(BitmapDescriptorFactory.fromResource(R.mipmap.ic_launcher))
            .title("$this ($typeString)"))!!


    override fun toString(): String {
        return if (Locale.getDefault().displayLanguage.contains("fr")) nameFr else name
    }
}
